pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -240,8 +241,12 @@ func (state PipelineRunState) getNextTasks(candidateTasks sets.String) []*Resolv
 }
 
 // IsStopping returns true if the PipelineRun won't be scheduling any new Task because
-// at least one task already failed or was cancelled in the specified dag
+// it was cancelled or stopped gracefully or at least one task already failed or was cancelled in the specified dag
 func (facts *PipelineRunFacts) IsStopping() bool {
+	if facts.SpecStatus == v1beta1.PipelineRunSpecStatusCancelledRunFinally ||
+		facts.SpecStatus == v1beta1.PipelineRunSpecStatusStoppedRunFinally {
```

Comment:
Instead of leveraging `stopping` for `CancelledRunFinally`, is it possible to not utilize `stopping` mode? How about introduce a `cancelledRunFinally` mode similar to `stopping`?

Reply:
Sure, makes sense.